test/unit/range/iterator_test_template.hpp

```diff
@@ -642,6 +651,69 @@ TYPED_TEST_P(iterator_fixture, compare_geq)
     }
 }
 
+// ---------------------------------------------------------------------------------------------------------------------
+// Contiguous Iterator
+// ---------------------------------------------------------------------------------------------------------------------
+
+TYPED_TEST_P(iterator_fixture, address_test)
+{
+    if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag>)
+    {
+        auto begin_it = std::ranges::begin(this->test_range);
+        EXPECT_EQ(std::to_address(begin_it), std::addressof(*begin_it));
+    }
+
+    if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag> &&
+                  TestFixture::const_iterable)
+    {
+        auto cbegin_it = std::ranges::cbegin(this->test_range);
+        EXPECT_EQ(std::to_address(cbegin_it), std::addressof(*cbegin_it));
+    }
+}
+
+template <typename it_begin_t, typename it_sentinel_t>
+inline void address_difference_test(it_begin_t it_begin, it_sentinel_t it_end)
+{
+    using difference_t = std::iter_difference_t<it_begin_t>;
+
+    for (auto it = it_begin; it != it_end; ++it)
+        EXPECT_EQ(std::to_address(it), std::to_address(it_begin) + static_cast<difference_t>(it - it_begin));
+}
+
+TYPED_TEST_P(iterator_fixture, address_difference)
+{
+    if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag>)
+    {
+        address_difference_test(std::ranges::begin(this->test_range), std::ranges::end(this->test_range));
+    }
+
+    if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag> &&
+                  TestFixture::const_iterable)
+    {
+        address_difference_test(std::ranges::cbegin(this->test_range), std::ranges::cend(this->test_range));
+    }
+}
+
+TYPED_TEST_P(iterator_fixture, address_to_end)
```

Comment:
```suggestion
TYPED_TEST_P(iterator_fixture, address_of_end)
```

Can you move this test after `address_of_begin`?

Reply:
This test is already after address_of_begin?

Reply:
Currently we have this order:

```c++
TYPED_TEST_P(iterator_fixture, address_of_begin)
TYPED_TEST_P(iterator_fixture, address_difference)
TYPED_TEST_P(iterator_fixture, address_of_end)
```

I wanted this order:

```c++
TYPED_TEST_P(iterator_fixture, address_of_begin)
TYPED_TEST_P(iterator_fixture, address_of_end)
TYPED_TEST_P(iterator_fixture, address_difference)
```

Reply:
ah now I get it :D